scripts/minikube-minishift-setup-env.sh

```diff
@@ -32,14 +34,75 @@ cp -avrf ./odo $GOBIN/
 shout "| Getting ginkgo"
 make goget-ginkgo
 
+
 case ${1} in
     minishift)
+        shout "| Setting environment for minishift"
         export MINISHIFT_ENABLE_EXPERIMENTAL=y 
         export PATH="$PATH:/usr/local/go/bin/"
         export GOPATH=$HOME/go
         mkdir -p $GOPATH/bin
         export PATH="$PATH:$(pwd):$GOPATH/bin"
-        sh .scripts/minishift-start-if-required.sh
+
+        #Export github API token to prevent pull limit error
+        export MINISHIFT_GITHUB_API_TOKEN=${MINISHIFT_GITHUB_API_TOKEN_VALUE}
+
+        # Verify if minishift or openshift are stopped and take appropriate actions
+        shout "| Checking if minishift needs to be installed..."
+        minishift version
+        msStatus=$(minishift status)
+        if [[ "$msStatus" == *"command not found"* ]]
+        then
+            shout "| Installing Minishift"
+            curl -Lo minishift.tgz https://github.com/minishift/minishift/releases/download/v1.34.3/minishift-1.34.3-linux-amd64.tgz
+            tar -xvzf minishift.tgz
+            sudo mv minishift-1.34.3-linux-amd64/minishift /usr/local/bin
+            rm minishift.tgz
+        fi
+
+        shout "| Checking if Minishift needs to be started..."
+        if [[ "$msStatus" == *"Does Not Exist"* ]] || [[ "$msStatus" == *"Minishift:  Stopped"* ]]
```

Comment:
Would you not need to read `msstatus` again, if the previous status was `command not found`? If the minishift command does not exist this will never get to start the vm

Reply:
will get msstatus again